File: 프로그래머스/lv2/42578. 의상/의상.js
function solution(clothes) {
    var answer = 1;
    let map = {};
    clothes.forEach(e=>{
        const [type,name] = e;
        name in map ? map[name]++ : map[name]=1
    })
    for(let key in map){
        answer *=(map[key]+1)
    }
    return answer-1;
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    dic = {}
    for i in clothes:
        a,b = i
        if b in dic:
            dic[b].append(a)
        else:
            dic[b] = [a]
    cnt = 1
    for k in dic:
        cnt*= len(dic[k])+1
    return cnt-1